libsplinter/src/network/connection_manager/auth/handlers.rs

```diff
@@ -0,0 +1,626 @@
+                    })?;
+            }
+            Ok(next_state) => panic!("Should not have been able to transition to {}", next_state),
+        }
+        Ok(())
+    }
+}
+
+/// Handler for the Authorization Error Message Type
+struct AuthorizationErrorHandler {
+    auth_manager: AuthorizationPoolStateMachine,
+}
+
+impl AuthorizationErrorHandler {
+    fn new(auth_manager: AuthorizationPoolStateMachine) -> Self {
+        AuthorizationErrorHandler { auth_manager }
+    }
+}
+
+impl Handler for AuthorizationErrorHandler {
+    type Source = ConnectionId;
+    type MessageType = authorization::AuthorizationMessageType;
+    type Message = authorization::AuthorizationError;
+
+    fn match_type(&self) -> Self::MessageType {
+        authorization::AuthorizationMessageType::AUTHORIZATION_ERROR
+    }
+
+    fn handle(
+        &self,
+        msg: Self::Message,
+        context: &MessageContext<Self::Source, Self::MessageType>,
+        _: &dyn MessageSender<Self::Source>,
+    ) -> Result<(), DispatchError> {
+        let auth_error = AuthorizationError::from_proto(msg)?;
+        match auth_error {
+            AuthorizationError::AuthorizationRejected(err_msg) => {
+                match self.auth_manager.next_state(
+                    context.source_connection_id(),
+                    AuthorizationAction::Unauthorizing,
+                ) {
+                    Ok(AuthorizationState::Unauthorized) => {
+                        info!(
+                            "Connection unauthorized by connection {}: {}",
+                            context.source_connection_id(),
+                            &err_msg
+                        );
+                    }
+                    Err(err) => {
+                        warn!(
+                            "Unable to handle unauthorizing by connection {}: {}",
+                            context.source_connection_id(),
+                            err
+                        );
+                    }
+                    Ok(next_state) => {
+                        panic!("Should not have been able to transition to {}", next_state)
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl MessageSender<ConnectionId> for AuthorizationMessageSender {
+    fn send(&self, id: ConnectionId, message: Vec<u8>) -> Result<(), (ConnectionId, Vec<u8>)> {
+        AuthorizationMessageSender::send(self, message).map_err(|msg| (id, msg))
+    }
+}
+
+impl Into<Box<dyn MessageSender<ConnectionId>>> for AuthorizationMessageSender {
+    fn into(self) -> Box<dyn MessageSender<ConnectionId>> {
+        Box::new(self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::collections::VecDeque;
+    use std::sync::{Arc, Mutex};
+
+    use protobuf::Message;
+
+    use crate::protos::authorization;
+    use crate::protos::network::{NetworkMessage, NetworkMessageType};
+
+    #[test]
+    fn connect_request_dispatch() {
```

Comment:
Tests should have a) a short description; b) a description of the purpose of the test; c) the procedure to achieve (b).

Reply:
As this was taken from the previous code, it did not originally have a comment, but will add one

Reply:
Fixed

Reply:
This comment was intended to apply to all the tests in the PR.

Reply:
These were the original comments, but I've updated them all to be more verbose. I probably wasn't clear in the commit message that this is the same code from the existing network auth code moved here.